### ISSUE TITLE
EOG Pass Improvements

### ISSUE THREAD

**Issue by maximiliankaul** [Fraunhofer-AISEC/cpg#1761]
- [ ] Doc URL to all nodes https://fraunhofer-aisec.github.io/cpg/CPG/specs/eog/
- [ ] `when` instead of the current `map`
- [ ] `createEOG()` -> `handle()`?
- [ ] `pushToEOG()` -> `connectWithCurrentPredecessors()`?
- [ ] `nextEdgeBranch` has to be set for branching nodes
- [ ] `nextEdgeBranch` does it work with nested `if`
- [ ] Add doc how to handle new nodes? `createEOG` / `pushToEOG` / `nextEdgeBranch` / ...

**Comment by konradweiss** [comment]
I used `attachToEOG` instead of `connectWIthCurrentPredecessors`.

**Comment by konradweiss** [comment]
Yes, `nextEdgeBranch` works with nested ifs, the EOG-edge of the outer if, when going from the root node to the first node in the Then-Statement will have branch value true. This first node would then be the first node in the condition of the inner IF. Only after the condition evaluation of the inner IF will there be an EOG-edge to the root node of the inner IF, and then to the inner Then-Statement that then has `nextEdgeBranch` set to true again

**Comment by oxisto** [comment]
Can we consider this closed?
